Library/Homebrew/language/python.rb

```diff
@@ -88,6 +88,15 @@ def self.setup_install_args(prefix)
       ]
     end
 
+    def self.rewrite_python_shebang(python_path)
+      Dir["**/*"].each do |f|
+        next unless File.file?(f)
+        next unless File.read(f).include?("/usr/bin/env python")
```

Comment:
Reading the whole file looks a bit too much probably, we could read the first several bytes and `next` if it's not `#!`

Reply:
```suggestion
        shebang_max_length = "#! /usr/bin/env python".length
        regex = %r{^#! ?/usr/bin/env python$}
        next unless regex.match?(f.read(shebang_max_length))
```

Reply:
@MikeMcQuaid space after `#!` is optional

Reply:
@bayandin Yup it's optional in the regex but used in full for the length calculation.